### PR TITLE
silo-vaults: renamed 'Morpho' to 'Silo'

### PR DESCRIPTION
**Pull request by IhorSF**

### CODE REVIEW

File: silo-vaults/contracts/MetaSilo.sol [hunk 1]
```diff
@@ -16,24 +16,25 @@ import {
...
+
 import {PendingUint192, PendingAddress, PendingLib} from "./libraries/PendingLib.sol";
 import {ConstantsLib} from "./libraries/ConstantsLib.sol";
 import {ErrorsLib} from "./libraries/ErrorsLib.sol";
 import {EventsLib} from "./libraries/EventsLib.sol";
 
-/// @title MetaMorpho
+/// @title MetaSilo
 /// @author Forked with gratitude from Morpho Labs.
 /// @custom:contact security@morpho.org
```
Comment (shadowyedd) [inline silo-vaults/contracts/MetaSilo.sol:35]: let's use silo email

File: silo-vaults/contracts/MetaSilo.sol [hunk 2]
```diff
@@ -16,24 +16,25 @@ import {
...
-/// @notice ERC4626 compliant vault allowing users to deposit assets to Morpho.
-contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorphoStaticTyping {
+/// @notice ERC4626 compliant vault allowing users to deposit assets to Silo.
```
Comment (shadowyedd) [inline silo-vaults/contracts/MetaSilo.sol:36]:
```suggestion
/// @notice ERC4626 compliant vault allowing users to deposit assets to any ERC4626 vault.
```

**Review by shadowyedd — APPROVED** [review]

File: silo-vaults/contracts/MetaSilo.sol (hunk 1 above)
Comment (IhorSF) [reply silo-vaults/contracts/MetaSilo.sol:35]: I opened a separate PR for this change https://github.com/silo-finance/silo-contracts-v2/pull/888

File: silo-vaults/contracts/MetaSilo.sol [hunk 3]
```diff
@@ -16,24 +16,26 @@ import {
     PendingUint192,
     PendingAddress,
     MarketAllocation,
-    IMetaMorphoBase,
-    IMetaMorphoStaticTyping
-} from "./interfaces/IMetaMorpho.sol";
+    IMetaSiloBase,
```
Comment (aiham02) [inline silo-vaults/contracts/MetaSilo.sol:19]: ISiloVaultsBase

File: silo-vaults/contracts/MetaSilo.sol [hunk 4]
```diff
@@ -16,24 +16,26 @@ import {
...
+    IMetaSiloStaticTyping
```
Comment (aiham02) [inline silo-vaults/contracts/MetaSilo.sol:20]: ISiloVaultsStaticTyping

File: silo-vaults/contracts/MetaSilo.sol [hunk 5]
```diff
@@ -16,24 +16,26 @@ import {
...
+} from "./interfaces/IMetaSilo.sol";
```
Comment (aiham02) [inline silo-vaults/contracts/MetaSilo.sol:21]: ISiloVaults

File: silo-vaults/contracts/MetaSilo.sol [hunk 6]
```diff
@@ -16,24 +16,26 @@ import {
...
 import {ConstantsLib} from "./libraries/ConstantsLib.sol";
 import {ErrorsLib} from "./libraries/ErrorsLib.sol";
 import {EventsLib} from "./libraries/EventsLib.sol";
 
-/// @title MetaMorpho
-/// @author Forked with gratitude from Morpho Labs.
-/// @custom:contact security@morpho.org
-/// @notice ERC4626 compliant vault allowing users to deposit assets to Morpho.
-contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorphoStaticTyping {
+/// @title MetaSilo
```
Comment (aiham02) [inline silo-vaults/contracts/MetaSilo.sol:33]: SiloVaults

File: silo-vaults/contracts/MetaSilo.sol [hunk 7]
```diff
@@ -16,24 +16,26 @@ import {
...
+/// @dev Forked with gratitude from Morpho Labs.
+/// @author Silo Labs
+/// @custom:contact security@silo.finance
+/// @notice ERC4626 compliant vault allowing users to deposit assets to Silo.
```
Comment (aiham02) [inline silo-vaults/contracts/MetaSilo.sol:37]: to Silo markets

File: silo-vaults/contracts/MetaSilo.sol [hunk 8]
```diff
@@ -16,24 +16,26 @@ import {
...
+contract MetaSilo is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaSiloStaticTyping {
```
Comment (aiham02) [inline silo-vaults/contracts/MetaSilo.sol:38]: SiloVaults

File: silo-vaults/contracts/MetaSilo.sol [hunk 9]
```diff
@@ -773,7 +775,7 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         }
 
         marketConfig.cap = _supplyCap;
-        // one time approval, so market can pull any amount of tokens from MetaMorpho in a future
+        // one time approval, so market can pull any amount of tokens from MetaSilo in a future
```
Comment (aiham02) [inline silo-vaults/contracts/MetaSilo.sol:778]: SiloVaults

File: silo-vaults/contracts/MetaSiloFactory.sol [hunk 10]
```diff
@@ -31,27 +32,27 @@ contract MetaMorphoFactory is IMetaMorphoFactory {
 
     /* EXTERNAL */
 
-    /// @inheritdoc IMetaMorphoFactory
-    function createMetaMorpho(
+    /// @inheritdoc IMetaSiloFactory
+    function createMetaSilo(
```
Comment (aiham02) [inline silo-vaults/contracts/MetaSiloFactory.sol:36]: createSiloVault

File: silo-vaults/deploy/MetaMorphoFactoryDeploy.s.sol [hunk 11]
```diff
@@ -3,37 +3,37 @@ pragma solidity 0.8.28;
...
     ETHERSCAN_API_KEY=$ARBISCAN_API_KEY FOUNDRY_PROFILE=vaults forge verify-contract \
-    0xdA72ab48AD4389B427b44d0dad393D5E5b209514 silo-vaults/contracts/MetaMorpho.sol:MetaMorpho \
+    0xdA72ab48AD4389B427b44d0dad393D5E5b209514 silo-vaults/contracts/MetaSilo.sol:MetaSilo \
     --chain 42161 --watch --compiler-version v0.8.28+commit.7893614a \
     --constructor-args <cast abi-encode output>
 
 */
-contract MetaMorphoFactoryDeploy is CommonDeploy {
-    function run() public returns (MetaMorphoFactory metaMorphoFactory) {
+contract MetaSiloFactoryDeploy is CommonDeploy {
```
Comment (aiham02) [inline silo-vaults/deploy/MetaMorphoFactoryDeploy.s.sol:27]: SiloVaultFactoryDeploy

File: silo-vaults/test/foundry/TimelockTest.sol [hunk 12]
```diff
@@ -75,18 +75,18 @@ contract TimelockTest is IntegrationTest {
         vault.submitTimelock(timelock);
     }
 
-    function testDeployMetaMorphoAboveMaxTimelock(uint256 timelock) public {
+    function testDeployMetaSiloAboveMaxTimelock(uint256 timelock) public {
```
Comment (aiham02) [inline silo-vaults/test/foundry/TimelockTest.sol:78]: testDeploySiloVaultAboveMaxTimelock

File: silo-vaults/.solhintignore [hunk 13]
```diff
@@ -1 +1 @@
-contracts/MetaMorpho.sol
+contracts/MetaSilo.sol
```
[the remark this answers is not in the document]
Comment (shadowyedd) [reply silo-vaults/.solhintignore:1]: Let's apply Aiham's comments but keep our internal policy for plurals. In this case let's keep SiloVault singular (no "s" at the end).

**Pull request merged**